[Zhu Le Yao] iP

ZhuLeYao opened the pull request:
# DukePro
> "I love DukePro :heart:" - *Lady Baba* [Source](https://en.wikipedia.org/wiki/Lady_Gaga)

DukePro helps you to record your tasks and events.
DukePro is:
- simple
- cleanly designed
- convenient

How to use DukePro:

1. Download it.
2. Double-click on it.
3. Add your tasks/ deadlines/ events.
4. DukePro will handle the rest! :tada:

Features:

- [x] Managing Tasks
- [x] Managing Deadlines
- [x] Managing Events

If you would like to practice yourself, here is the `main` method:
```java
public class Main {
    public static void main(String[] args) {
        Application.launch(MainApp.class, args);
    }
}
```

Commit:
Add Gradle support

ZhuLeYao committed:
Add skeletal version of Duke

ZhuLeYao committed:
Add list to store text entered by user

ZhuLeYao committed:
Revert "Add list to store text entered by user"

This reverts commit 2a7c67198fcfd91557638d4bf8e5bc3cb35a7632.

ZhuLeYao committed:
Add list to store text entered by user

ZhuLeYao committed:
Add option to mark tasks and Task class

Tasks can be marked as done or undone. Wordings are modified.

Task class is added to store name, number and task information.

ZhuLeYao committed:
Add task types: Todo, Deadline and Event

Subclasses of Tasks are created, namely Todo, Deadline and Event.
Deadline and Event are able to store information on the task's time.
Some wording are also modified.

ZhuLeYao committed:
Add exceptions to task commands

Exceptions are missing index, invalid index, invalid command and
incomplete command caused by missing task or task time.

ZhuLeYao committed:
Add function to delete a task

ZhuLeYao committed:
Add save task list to hard disk function

Existing ask list is able to be reflected in duke when restarted.

ZhuLeYao committed:
Fix time to be read as date time instead of string

Time format is shown as "MMM dd yyyy HHmm a".

Enigmatrix commented on src/main/java/Deadline.java:
```diff
@@ -0,0 +1,58 @@
+    }
+
+    @Override
+    public void markAsDone() {
```
Good splitting of lines!

Enigmatrix commented on src/main/java/Duke.java:
```diff
@@ -1,10 +1,254 @@
+
+            try {
+                if (command.equals("list")) {
+                    this.printCommandList(allTasks);
```
unnecessary qualifier to `this`

Enigmatrix commented on src/main/java/Task.java:
```diff
@@ -0,0 +1,66 @@
+        this.taskNumber = taskNumber;
+        this.taskStatus = taskStatus;
+        this.task = task;
+        this.totalNumOfTasks = totalNumOfTasks;
```
this field is unused

Enigmatrix commented on src/main/java/Task.java:
```diff
@@ -0,0 +1,66 @@
+public class Task {
```
this can be made into a abstract class, with abstract method implementations

Enigmatrix commented on src/main/java/Deadline.java:
```diff
@@ -0,0 +1,58 @@
+public class Deadline extends Task {
```
Needs Javadocs for these public classes & public methods

ZhuLeYao committed:
Merge branch 'branch-level-7'

# Conflicts:
#	src/main/java/Deadline.java
#	src/main/java/Duke.java
#	src/main/java/Event.java
#	src/main/java/Task.java

ZhuLeYao committed:
Remove print lines for debugging

ZhuLeYao committed:
Update git ignore list

ZhuLeYao committed:
Merge branch 'branch-level-8'

# Conflicts:
#	src/main/java/Deadline.java
#	src/main/java/Duke.java
#	src/main/java/Event.java
#	src/main/java/Task.java

ZhuLeYao committed:
Edit data time format of task saved to hard disk

ZhuLeYao committed:
Add command to find deadlines or events on date

ZhuLeYao committed:
Move exception methods from Duke to DukeException

ZhuLeYao committed:
Refactor the code to extract out closely related code as classes

Ui: deals with interactions with the user
Storage: deals with loading tasks from the file and saving tasks in the
file
Parser: deals with making sense of the user command
TaskList: contains the task list e.g. it has operations to add/delete
tasks in the list

ZhuLeYao committed:
Divide classes into packages

ZhuLeYao committed:
Merge branch 'add-gradle-support'

ZhuLeYao committed:
Add JUnit tests for Todo and TaskList

ZhuLeYao committed:
Package the App as a JAR file

ZhuLeYao committed:
Add JavaDoc comments

ZhuLeYao committed:
Tweak the code to comply with a coding standard

ZhuLeYao committed:
Add function for users to find a task by searching for a keyword

ZhuLeYao committed:
Merge branch 'branch-A-JavaDoc'

ZhuLeYao committed:
Merge branch 'branch-A-CodingStandard'

# Conflicts:
#	src/main/java/duke/exception/DukeException.java
#	src/main/java/duke/task/Deadline.java
#	src/main/java/duke/task/Event.java
#	src/main/java/duke/task/Task.java

ZhuLeYao committed:
Merge branch 'branch-Level-9'

# Conflicts:
#	src/main/java/duke/main/Ui.java

ZhuLeYao committed:
Update JavaDoc comments for duke.main package

ZhuLeYao committed:
Update JavaDoc comments for duke.task package

ZhuLeYao committed:
Update JavaDoc comments for duke.exception package

ZhuLeYao committed:
Create GUi for Duke

ZhuLeYao committed:
Remove border lines from Ui

ZhuLeYao committed:
Edit ShadowJar values

ZhuLeYao committed:
Merge branch 'branch-Level-10'

ZhuLeYao committed:
Add Assertions for assumptions in main

ZhuLeYao committed:
Modify git.ignore to allow ACTUAL.TXT

ZhuLeYao committed:
Improve code quality

In Parser.java, "mark", "unmark" and "delete" commands has similar parsing codes.
Similar areas are taken out to form method
checkExceptionAndGetTask(String, TaskList).

In Storage.java, "mark" and "unmark" commands has similar file reading
procedures.
Similar areas are taken out to form method
markOrUnmarkStorage(File, String, Task, DateTimeFormatter, TaskList,
String).
"delete" command's file reading procedure is taken out to form method
deleteStorage(File, String, TaskList).

ZhuLeYao committed:
Merge pull request #2 from ZhuLeYao/branch-A-Assertions

Add Assertions for assumptions in main

ZhuLeYao committed:
Merge branch 'master' of github.com:ZhuLeYao/ip

ZhuLeYao committed:
Merge pull request #3 from ZhuLeYao/branch-A-CodeQuality

Improve code quality

ZhuLeYao committed:
Merge branch 'master' of github.com:ZhuLeYao/ip

ZhuLeYao committed:
Add a way to archive items

A way to archive every task and action is added. Tasks archived will be
saved in archive.txt in the folder "data". Deleted tasks will be
removed from tasks.txt but remains stored in archive.txt. User can use
command "archive" to archive a copy of the current tasks in list. This
will be saved in archiveAll.txt in folder "data". archive.txt will still
contain history of all tasks, not just tasks in current list.

ZhuLeYao committed:
Merge branch 'branch-BCD-Extension'

* branch-BCD-Extension:
  Add a way to archive items

A way to archive every task and action is added. Tasks archived will be
saved in archive.txt in the folder "data". Deleted tasks will be
removed from tasks.txt but remains stored in archive.txt. User can use
command "archive" to archive a copy of the current tasks in list. This
will be saved in archiveAll.txt in folder "data". archive.txt will still
contain history of all tasks, not just tasks in current list.

ZhuLeYao committed:
Add Ui image and update README.md

ZhuLeYao committed:
Add images into README.md

ZhuLeYao committed:
Format README.md

ZhuLeYao committed:
Format README.md

ZhuLeYao committed:
Edit boolean naming of isMarked

ZhuLeYao committed:
Correct archive saving mechanism

ZhuLeYao committed:
Abstract long methods
